### PR TITLE
Recurring events

### PR DESCRIPTION
**Pull request by olsona**
Building a class for recurring events in calendar.

### CODE REVIEW

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package org.dataportabilityproject.types.transfer.models.calendar;
+
+// This interface could be used by other models.  If so, we should move it out and make it more
+// broadly useful.
+interface Property<T> {
+  T getValue();
+}
+
+class RuleDay implements Property<RecurringEvent.Day> {
+  private RecurringEvent.Day day;
```
Comment (holachuy) [inline portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java]: final here and for the other implementation classes below

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  private boolean isNegative;
+  private RuleNumber number;
+  private RuleDay day;
+
+  ByDay(boolean isNegative, RuleNumber number, RuleDay day) {
+    this.isNegative = isNegative;
+    this.number = number;
+    this.day = day;
+  }
+}
```
Comment (holachuy) [inline portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java]: new line at the end (IntelliJ has this in settings on save)

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package org.dataportabilityproject.types.transfer.models.calendar;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class RecurringEvent {
+
+  public class RRule {
+    Freq freq;
```
Comment (holachuy) [inline portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java]: final on all these member variables

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  enum Day {SU, MO, TU, WE, TH, FR, SA}
+
+  enum TimeType {
+    DATETIME("DATE-TIME"),
+    DATE("DATE"),
+    PERIOD("PERIOD"),
+    UNDEFINED("UNDEFINED");
+
+    private String title;
```
Comment (holachuy) [inline portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java]: final

Comment (olsona) [reply portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java]: Done.

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java (hunk 3 above)
Comment (olsona) [reply portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/RecurringEvent.java]: Done.

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java (hunk 2 above)
Comment (olsona) [reply portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java]: Done.  (Did not have that setting on my Mac, thanks for reminding me!)

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java (hunk 1 above)
Comment (olsona) [reply portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java]: Done.

File: portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package org.dataportabilityproject.types.transfer.models.calendar;
+
+// This interface could be used by other models.  If so, we should move it out and make it more
+// broadly useful.
+interface Property<T> {
+  T getValue();
+}
+
+final class RuleDay implements Property<RecurringEvent.Day> {
+  private RecurringEvent.Day day;
```
Comment (holachuy) [inline portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java]: This member should be final also as well as the ones below in this class.

Comment (olsona) [reply portability-types-transfer/src/main/java/org/dataportabilityproject/types/transfer/models/calendar/Property.java]: Done.  Caught another error too - ByDay needs getters.

**Pull request merged**